src/Visualization/Python/PlotTrajectories.py

```
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Plot trajectory from inspiral run, concatenating all segments
+
+import click
+import h5py
+import matplotlib.pyplot as plt
+import numpy as np
+
+from spectre.Visualization.Plot import (
+    apply_stylesheet_command,
+    show_or_save_plot_command,
+)
+
+
+@click.command(name="trajectories")
+@click.argument(
+    "h5_files",
+    nargs=-1,
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, readable=True),
+)
+@click.option(
+    "--subfile-name-aha",
+    "-A",
+    default="ApparentHorizons/ControlSystemAhA_Centers.dat",
+    help=(
+        "Name of subfile containing the apparent horizon centers for object A."
+    ),
+)
+@click.option(
+    "--subfile-name-ahb",
+    "-B",
+    default="ApparentHorizons/ControlSystemAhB_Centers.dat",
+    help=(
+        "Name of subfile containing the apparent horizon centers for object B."
+    ),
+)
+@show_or_save_plot_command()
+@apply_stylesheet_command()
+def plot_trajectories_command(h5_files, subfile_name_aha, subfile_name_ahb):
+    """Plot trajectories in inspiral simulation
+
+    Concatenates partial trajectories from each h5 reductions files and plots
+    full trajectories.
+    """
+    reductionfiles = h5_files
+    AhA, AhB = import_A_and_B(
+        reductionfiles, subfile_name_aha, subfile_name_ahb
+    )
+    if AhA is not None and AhB is not None:
+        plot_trajectory(AhA, AhB)
+    else:
+        click.echo("Error: Unable to import data or not valid data.")
+
+
+def import_A_and_B(filenames, subfile_name_aha, subfile_name_ahb):
+    A_data = []
+    B_data = []
+
+    for filename in filenames:
+        with h5py.File(filename, "r") as file:
+            if subfile_name_aha in file:
```

Remove these `if` conditions as well. It's great that you're considering what happens when errors happen! I agree that's very important. You always want to give the user enough information and guidance. In this case just access the subfile so an error is thrown when it doesn't exists. That should be enough, and it's better than silently ignoring the file.